### DISCUSSION TITLE
Postgresql receiver & Prometheus: how to get back database and table as attributes?

### DISCUSSION THREAD

**Discussion by unwork-ag** [open-telemetry/opentelemetry-collector-contrib#18457] [Q&A] [answered]
According to the documentation of the postgresql receiver the database and table name attributes are no longer available as metric attributes, instead they are now part of the resource attributes. However, for using these metrics through Prometheus, I need access to these attributes and as far as I can see resource attributes are not accessible through Prometheus.
I already tried to transform the resource attributes to metric attributes using the attributes processor but so far I had no success (I used from_attribute and from_context - maybe I need to prefix the key to indicate to take the value from the resource attributes?).

In essence: how would I use metrics data from this receiver in Prometheus (when I want to filter for particular tables etc.)?

**Comment by atoulme** [comment, marked as answer]
Resource attributes are usually part of the export. They are just the same across all the data points, so we bundle metrics by resource. When exporting to Prometheus, those attributes should be inlined in the export. However, it looks like they're not exported by default. See the flag `resource_to_telemetry_conversion` to turn that on.

**Reply by unwork-ag** [reply]
Thanks! Where would I apply this flag? Is that part of the Prometheus exporter config?

**Reply by atoulme** [reply]
Here is the doc reference: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/prometheusexporter/README.md?plain=1#L23
